win32/src/win32evtlog.i

```diff
@@ -579,7 +578,7 @@ static PyObject *FormatMessageInternal(EVT_HANDLE metadata, EVT_HANDLE event, DW
 	}
 
 	if (flags == EvtFormatMessageKeyword) {
-		buf[returned_size * sizeof(WCHAR)] = L'\0';
+		buf[returned_size - 1] = L'\0';
```

Comment:
This doesn't make sense to me - this is treating `returned_size` as the number of bytes and not chars, which is the opposite of what we need to do. IOW, this will be placing a \0 in the middle of the allocated buffer not at the end.

Reply:
According to docs https://docs.microsoft.com/en-us/windows/win32/api/winevt/nf-winevt-evtformatmessage

> _BufferUsed_
The size, in **characters** of the caller-allocated buffer that the function used or the required buffer size if the function fails with ERROR_INSUFFICIENT_BUFFER.


so the `returned_size ` variable, which is given as a `BufferUsed` parameter, should be an amount of characters, shouldn't it?

Without those changes, our app had been constantly crashing, and that's(as I assume) why:
* `EvtFormatMessage` sets some value for the `returned_size`, for example `10` characters.

* Then we allocate memory for 10 **bytes**, not characters.
_This is a previous code._
```
	allocated_size *= sizeof(WCHAR);
	buf = (WCHAR *)malloc(allocated_size);
```
So we have a buffer with a size - 10 bytes, but the size of the string is 10 wide characters(20 bytes)
* Then:
_This is a previous code._
```
buf[returned_size * sizeof(WCHAR)] = L'\0';
```
we try to set zero termination character by index 20 in the buffer with size 10 bytes. The result - segfault.

It also seems like we should `returned_size - 1`. You can see that in the official [docs](https://docs.microsoft.com/en-us/windows/win32/wes/formatting-event-messages) too.

Reply:
We should not have to do  `returned_size - 1`, because we increment `allocated_size`, but I am still sure that we should not multiply `returned_size` to `sizeof(WCHAR)`. 

According to docs https://docs.microsoft.com/en-us/windows/win32/api/winevt/nf-winevt-evtformatmessage

>BufferUsed
The size, in characters of the caller-allocated buffer that the function used or the required buffer size if the function fails with ERROR_INSUFFICIENT_BUFFER.

so the `returned_size` variable, which is given as a `BufferUsed` parameter, should be a number of characters, shouldn't it?

Considering that, we don't have to multiply indices because the type of the `buf` variable is `LPWSTR`, not just a raw bytes array. If we do multiply, we go out of bufffer's bounds. Sometimes it does not crash because `malloc` may allocate more bytes but  this is still undefined behaviour.